content/english/java-tictactoe/activity-5.md

```diff
@@ -3,5 +3,80 @@ title: "5. (Optional) AI Move"
+{{% /notice %}}
+
+### Let's make the computer think
+Previously, we use a `Random` object to randomly generate a move for the computer, making the competitiveness of the computer is pretty low.
+
+In this exercise, we want to increase the difficulty of the game by making optimal decisions for the computer.
+
+We will do so by adding artificial intelligence into our program with the use the <b>Minimax</b> algorithm (a well-defined procedure that allows computers to solve problems).
+
+### Minimax Algorithm
+Minimax is an algorithm that is often used in two-player games to make optimal decisions for a player. 
+
+- The algorithm makes optimal decision for the maximizer by examining all the possible future game states based on the current game board. 
+- If there is a winner or tie, the game state will be given a score. (Score: computer wins (`positive value`), player wins (`negative value`), tie (`0`)).
+- The two players are labeled as <b>maximizer</b> and <b>minimizer</b> respectedly. In this case, the computer is the maximizer, and the player is the minimizer. 
+- The maximizer tries to get the highest score possible, and the minimizer tries to minimize the score of its opponent. 
+
+<img src="../images/minimax.png" height="500"/> 
+
+During the minimizing round, we pick the game state with the smallest score. And, in the maximizing round, we pick the game state with the largest score.
+
+If you follow along through the maximizing/minimizing rounds on the picture above, you will note that the best move for the computer is to place `"X"` at position 8.
+
+### Write the Method `getComputerMoveAI()`
+In activity-3, you wrote the method `int getComputerMove(String[] curBoard)` to randomly generate a spot for the computer. Let's write another method called `getComputerMoveAI(String[] curBoard)` that returns the optimal move for the computer. 
+
+We will write this method assuming we have the `int minimax(String[] curBoard, boolean isMaximizing)` method implemented, which you will write later.
+
+1. For each available spot on the game board, place `"X"` at that spot and get the score for that board by calling `minimax()`. 
+{{% notice hint %}}
+- Note that we call the method `minimax()` inside of the implementation of itself. We call a function that calls itself during its execution a <b>recursive function</b>.
+- Note you should pass `false` for the second argument since it'd be the minimizer's turn.
+- You should change that spot back to `"O"` after getting the score, so you maintain the original state of the game board during the next iteration.
+{{% /notice %}}
+
+2. Track the largest score and its corresponding position of the board at each iteration. Return the position with the largest score.
+{{% notice hint %}}
+- Have a variable called `bestScore` that stores the current best score and have an initial value of `Integer.MIN_VALUE` (minimum value of an integer).
+{{% /notice %}}
```

Comment:
Consider using code snippets to explain some of the stuff here. The students still don't know the entire program and have to stitch these up anyway so that is still a challenge

Reply:
Added!